src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './types/Todo';
+
+const preparedTodos: PreparedTodo[] = todosFromServer.map(todo => {
+  const user = usersFromServer.find(({ id }) => id === todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const [todos, setTodos] = useState(preparedTodos);
+
+  function handleTitleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  }
+
+  function handleUserIdSelect(event: React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  }
+
+  function handleAdd(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    const user = usersFromServer.find(({ id }) => id === userId);
+    const todoIds = todos.map(({ id }) => id);
+    const maxId = Math.max(...todoIds);
+
+    const newTodo = {
+      id: maxId + 1,
+      title,
+      completed: false,
+      userId,
+      user,
+    };
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
```

Comment:
Better to do this check at the beginning fo the function, so you will avoid useless calculations if no needed